src/datapilot/clients/altimate/client.py

```diff
@@ -0,0 +1,81 @@
+import logging
+
+import requests
+
+
+class APIClient:
+    def __init__(self, api_token="", base_url="", tenant=""):
+        self.api_token = api_token
+        self.base_url = base_url
+        self.tenant = tenant
+        self.logger = logging.getLogger(self.__class__.__name__)
+
+    def _get_headers(self):
+        headers = {
+            "Content-Type": "application/json",
+        }
+
+        if self.api_token:
+            headers["Authorization"] = f"Bearer {self.api_token}"
+
+        if self.tenant:
+            headers["x-tenant"] = self.tenant
+
+        return headers
+
+    def log(self, message):
+        self.logger.debug(message)
+
+    def get(self, endpoint, params=None, timeout=None):
+        url = f"{self.base_url}{endpoint}"
+        headers = self._get_headers()
+
+        self.logger.debug(f"Sending GET request for tenant {self.tenant} at url: {url}")
+        response = requests.get(url, headers=headers, params=params, timeout=timeout)
+        self.logger.debug(f"Received GET response with status: {response.status_code }")
+        return response.json() if response.status_code == 200 else None
+
+    def post(self, endpoint, data=None, timeout=None):
+        url = f"{self.base_url}{endpoint}"
+        headers = self._get_headers()
+
+        self.logger.debug(f"Sending POST request for tenant {self.tenant} at url: {url}")
+        response = requests.post(url, headers=headers, json=data, timeout=timeout)
+        self.logger.debug(f"Received POST response with status: {response.status_code }")
+
+        return response
```

Comment:
We should handle exceptions and not 200 errors nicely. We would want to propogate errors wherever possible to the User

Reply:
Potentially can do the error handling at the individual endpoint level, since you know all the possible errors

Reply:
Yeah, the error will propagate to the users. I removed the additional logging that I was doing based on status code